File: sast-go/cases/accuracy/context_sensitive/argument_return_value_passing/return_value_passing_002_T/return_value_passing_002_T.go
// evaluation information start
// real case = true
// evaluation item = 准确度->上下文敏感分析->参数/返回值传递
// scene introduction = 
// level = 2
// bind_url = accuracy/context_sensitive/argument_return_value_passing/return_value_passing_002_T/return_value_passing_002_T
// evaluation information end

package main
import "os/exec"

func return_value_passing_002_T(__taint_src interface{}) {
	data := process(__taint_src)
	__taint_sink(data)
}

func __taint_sink(o interface{}) {
	_ = exec.Command("sh", "-c", o.(string)).Run()
	}

func process(__taint_src interface{}) interface{} {
	return __taint_src
}

func main() {
    __taint_src := "taint_src_value"
    return_value_passing_002_T(__taint_src)
}

File: sast-python3/case/completeness/single_app_tracing/function_call/return_value_passing/return_value_passing_002_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = true
# evaluation item = 完整度->单应用跟踪完整度->函数和方法调用->返回值传递
# scene introduction = 普通
# level = 2
# bind_url = completeness/single_app_tracing/function_call/return_value_passing/return_value_passing_002_T
# evaluation information end
import os


def return_value_passing_002_T(taint_src):
    data = process(taint_src)
    taint_sink(data)


def process(src):
    return src


def taint_sink(o):
    os.system(o)


if __name__ == '__main__':
    taint_src = "taint_src_value"
    return_value_passing_002_T(taint_src)

